conftest.py

```diff
@@ -4,11 +4,38 @@
 import requests
 from typing import Callable
 import logging
+from dataclasses import dataclass
 
 
 log = logging.getLogger(__name__)
 
 
+@dataclass
+class SensorInfo:
+    name: str
+    hid: str
+    model: str
+    firmware_version: int
+    reading_interval: int  # RuntimeError: Sensor didn't reset to factory property if float
+
+
+    def __post_init__(self):
+        if not isinstance(self.name, str) or self.name == "":
+            raise TypeError("'name' is not a string or empty")
+
+        if not isinstance(self.hid, str) or self.hid == "":
+            raise TypeError("'hid' is not a string or empty")
+
+        if not isinstance(self.model, str) or self.model == "":
+            raise TypeError("'model' is not a string or empty")
+
+        if not isinstance(self.firmware_version, int) or self.firmware_version == "":
```

Comment:
Testing for a non-empty string is suitable for string properties but here we've got an integer. We'd like to test this for validity from sensor's standpoint. That would be checking that the firmware version is between 10 and 15, both ends included. You could test this with the following code:
```
10 <= self.firmware_version <= 15
```
Of course, you'll need to update the error message in the `ValueError` to reflect this new check.